### ISSUE TITLE
UnboundLocalError: local variable 'vals' referenced before assignment (in append method of mdf4)

### ISSUE THREAD

**Issue by JulienGrv** [danielhrisca/asammdf#24]
Hi Daniel,

vals is not assigned, what is it supposed to be here?

```plain
Traceback (most recent call last):
  File "test.py", line 44, in <module>
    main()
  File "test.py", line 32, in main
    mdf_reduced = mdf.filter(channels)
  File "D:\Workspace\asammdf\asammdf\mdf.py", line 543, in filter
    common_timebase=True,
  File "D:\Workspace\asammdf\asammdf\mdf4.py", line 2142, in append
    types.append(vals.dtype)
UnboundLocalError: local variable 'vals' referenced before assignment
```

**Comment by danielhrisca** [comment]
The field type is appended in the previous line (2141) so you should comment out the line 2142.

**Comment by JulienGrv** [comment]
It seems to be working now. I push this change as well

**Comment by JulienGrv** [comment]
Fixed in PR #25
